server/modules/syllabus/commands/create-syllabus/create-syllabus-dto.ts

```diff
@@ -0,0 +1,80 @@
+import { IsArray, IsEmail, IsString, MaxLength, IsNotEmpty, IsOptional, validateOrReject } from 'class-validator'
+import { Syllabus } from '../../domain/syllabus'
+import { SyllabusCourse } from '../../domain/syllabus-course'
+import { SyllabusCode, generate } from '../../domain/syllabus-code'
+
+export type CreateSyllabus = Omit<Syllabus, 'id'>
+
+export class CreateSyllabusDto implements CreateSyllabus {
+  @MaxLength(255)
+  @IsString()
+  @IsNotEmpty()
+  readonly name!: string
+
+  @MaxLength(64)
+  @IsString()
+  @IsNotEmpty()
+  readonly instructor!: string
+
+  @MaxLength(255)
+  @IsString()
+  @IsNotEmpty()
+  readonly location!: string
```

Comment:
@korgan00 it looks like these 3 fields (`name`, `instructor`, `location`) are required and the rest of the fields are optional. was there a discussion on what should be required and what should be optional? and if so can the UI be updated to indicate required fields.

in addition, when there is a server error (e.g., you don't fill out a required field), it is not surfaced in the UI and it looks like the `Publish` button does work (i.e., nothing is happening). there is no indication of a failure. 

checking the network tab you do see POST request but it fails and there is a failure response object.

@pandasa123 @JRussellHuffman do you have design and/or thoughts on what fields should be required and how it should be surfaced in the UI?

Reply:
@korgan00 and me had a discussion about what fields should be required or not. How we didn't know initially what we would can find or how the users will use the tool we decided the to put the minimum fields required as possible (initially `name`, `instructor` and `location`).

About the errors @vabarbosa you are totally right. Right now I'm not managing correctly the errors (mostly those one related with the validations) and is an improvement that I need to implement. What we talked @korgan00 and me was finish the main flow and after that try to fix these final things. In this case validate the input correctly and show to the user the feedback for those fields that need to be fixed.

Reply:
Btw, as a side note, the `instructor` is being filled with the user that creates the syllabus so it's automatically filled. The required parameter can be removed but we thought that it was a good idea to specify that that parameter must be always present in the Syllabus. We can have a discussion about it too, we were not totally sure about these decisions so really glad to hear more opinions.

Reply:
The following fields should be required:

- Name
- Instructor

I feel like we should add a `* required` above the input field (or in the placeholder)

Reply:
Just wanted to note that the `:required="true"` attributes _are_ included in the `<bx-input>` components, for example, shown on this [line here](https://github.com/Qiskit/platypus/pull/855/files#diff-e692348f820ccb24cddf18f987ae8ca9971c3d9f9245e95c27b78797915caf59R12)

Also, I believe this is _how_ that specific attr should be used, according to the Carbon web component's properties shown [here](https://web-components.carbondesignsystem.com/?path=/docs/components-input--form-item)... I tried adding a `required-validity-message` property locally, but nothing happens still. (Maybe we need to add our own validation check?)

Reply:
Let's remove the requirement for the `location` param then. But yeah, as @pandasa123 mentions above could be a good idea to add a `*` to the input to specify that they are required.

About the validation messages @techtolentino, I'm don't have ready a good way for you to check them yet. I have a conversation pending with @korgan00 and you about how we can do this but I need to finish first the refactor sprint. We can have a meeting for the next sprint if you see it ok!

Reply:
the carbon component can check required fields, you need to call `checkValidity()` on the field: https://github.com/Qiskit/platypus/pull/928

Reply:
Oh yeah, sorry, when I said that I didn't have ready a way to check the validations I was thinking from backend to frontend. For only a frontend validation that's enough, totally.